question10/main.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+// loading img 
+const wrapper = document.getElementById('wrapper');
+const loadImg = document.createElement('img');
+const showLoadImg = () => {
+    loadImg.src = "loading-circle.gif";
+    wrapper.appendChild(loadImg);
+} 
+
+const contentsList = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+                      {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
+
+
+const promise = new Promise((resolve) => {
+    setTimeout(() =>{
+      resolve(contentsList);
+    }, 3000)
+})
+
+async function asyncFunc() {
+    showLoadImg();
+    try {
+        const result = await promise;
+        createList(result);
+    }
+    catch(error) {
+        console.error(error);
+    }
+    finally{
+        loadImg.remove()
+    }
+}
+asyncFunc();
+
```

Comment:
I just added try-catch-finally to async function simply.
①Loading img is showed.
②try: Here is where the exception might occur.
③catch: If there are some problems, Error is showed.
④finally: This is always done after the try and catch blocks have completed their execution.(In this case, the loading image will be removed.)

Reply:
I think you're right.